ang/crmMailing/Templates.js

```diff
@@ -0,0 +1,134 @@
+(function(angular, $, _) {
+  // example <select crm-mailing-templates crm-mailing="mymailing"></select>
+  angular.module('crmMailing').directive('crmMailingTemplates', function(crmUiAlert) {
+      return {
+          restrict: 'AE',
+          require: 'ngModel',
+          scope: {
+            ngRequired: '@'
+          },
+          templateUrl: '~/crmMailing/Templates.html',
+          link: function(scope, element, attrs, ngModel) {
+            scope.template = ngModel.$viewValue;
+
+            var refreshUI = ngModel.$render = function refresuhUI() {
+              scope.template = ngModel.$viewValue;
+              if (ngModel.$viewValue) {
+                $(element).select2('val', ngModel.$viewValue);
+              }
+            };
+
+            // @return string HTML representing an option
+            function formatItem(item) {
+              if (!item.id) {
+                // return `text` for optgroup
+                return item.text;
+              }
+              return '<span class="crmMailing-template">' + item.text + '</span>';
+            }
+
+            var rcpAjaxState = {
+              input: '',
+              entity: 'civicrm_msg_templates',
+              type: 'include',
+              page_n: 0,
+              page_i: 0,
+            };
+
+            $(element).select2({
+              width: '36em',
+              placeholder: "<i class='fa fa-clipboard'></i> Mailing Templates",
+              formatResult: formatItem,
+              escapeMarkup: function(m) {
+                return m;
+              },
+              multiple: false,
+              initSelection: function(el, cb) {
+
+                  var value = el.val();
+
+                  CRM.api3('MessageTemplate', 'getlist', { params: { id: value }, label_field: 'msg_title' }).then(function(tlist) {
+
+                      var template = {};
+
+                      if (tlist.count) {
+                        $(tlist.values).each(function(id, val) {
+                          template.id = val.id;
+                          template.text = val.label;
+                        })
+                      }
+
+                      cb(template);
+                  });
+              },
+              ajax: {
+                  url: CRM.url('civicrm/ajax/rest'),
+                  quietMillis: 300,
+                  data: function(input, page_num) {
+                    if (page_num <= 1) {
+                      rcpAjaxState = {
+                        input: input,
+                        entity: 'civicrm_msg_templates',
+                        type: 'include',
```

Comment:
probably unneeded here

Reply:
Agreed, this was only needed for grouping the inclusion of recipients/mailing groups.